src/app/Utilities/FormBuilderUtilities.ts

```diff
@@ -0,0 +1,139 @@
+/* Import Utilities */
+import KeycloakService from "keycloak/Keycloak";
+
+/* Import Types */
+import { SourceSystem, Mapping, EditTarget, Dict } from "app/Types";
+
+/* Import API */
+import GetSourceSystem from "api/sourceSystem/GetSourceSystem";
+import InsertSourceSystem from 'api/sourceSystem/InsertSourceSystem';
+import PatchSourceSystem from 'api/sourceSystem/PatchSourceSystem';
+import GetMapping from "api/mapping/GetMapping";
+import InsertMapping from 'api/mapping/InsertMapping';
+import PatchMapping from 'api/mapping/PatchMapping';
+
+
+const DefineEditTarget = async (targetName: string, id: string) => {
+    let editTarget: EditTarget = {};
+    let copyEditTarget = { ...editTarget };
+
+    if (targetName === 'sourceSystem') {
+        await GetSourceSystem(id).then(async (sourceSystem) => {
+            if (sourceSystem) {
+                copyEditTarget = { ...copyEditTarget, sourceSystem: { ...sourceSystem, id } };
+
+                await GetMapping(sourceSystem.mappingId).then((mapping) => {
+                    if (mapping) {
+                        copyEditTarget = { ...copyEditTarget, ...mapping };
+                    }
+                }).catch(error => {
+                    console.warn(error);
+                });
+            }
+        }).catch(error => {
+            console.warn(error);
+        });
+    } else if (targetName === 'mapping') {
+        await GetMapping(id).then((mapping) => {
```

Comment:
Actually, the part in Source System is not needed, since we do not allow a mapping to be edited when editing a Source System, since other Source Systems may also be using that particular mapping.
Remove the GetMapping call in the SourceSystem part